modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-email-otp-web/src/main/java/com/liferay/multi/factor/authentication/email/otp/web/internal/checker/EmailOTPBrowserMFAChecker.java

```diff
@@ -148,11 +148,13 @@ public boolean verifyBrowserRequest(
 						"nonexistent user " + userId);
 			}
 
-			_routeAuditMessage(
-				_mfaEmailOTPAuditMessageBuilder.
-					buildNonexistentUserVerificationFailureAuditMessage(
-						CompanyThreadLocal.getCompanyId(), userId,
-						_getClassName()));
+			if (_mfaEmailOTPAuditMessageBuilder != null) {
```

Comment:
@slnn When you do null check for volatile fields, you should

```
XXX xxx = _xxx;

if (xxx != null) {
    // do something with xxx
}
```

Reply:
Please see https://github.com/liferay-core-infra/liferay-portal/pull/4773